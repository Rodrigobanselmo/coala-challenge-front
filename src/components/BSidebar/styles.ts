import { css } from "@emotion/react";
import { styled, Stack, StackProps } from "@mui/material";

interface StyledStackPros extends StackProps {
  is_close: number;
}
export const StyledFlex = styled(Stack)<StyledStackPros>`
  height: 100%;
  background-color: ${({ theme }) => theme.palette.sidebar.background};
  width: 18rem;
  transition: all 0.5s ease-in-out;
  display: flex;
  position: relative;

  ${(props) =>
    props.is_close &&
    css`
      width: 4rem;
    `}
`;
