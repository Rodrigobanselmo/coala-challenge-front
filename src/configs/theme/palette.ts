const palette = {
  primary: {
    light: "#4071d6",
    main: "#2153b7",
    dark: "#2153b7",
    contrastText: "#fff",
  },
  secondary: {
    extraLight: "#fca557",
    light: "#f28338",
    main: "#F27329",
    dark: "#d9560b",
    contrastText: "#fff",
  },
  sidebar: {
    background: "#1A202C",
  },
  mainBlur: {
    90: "rgba(242, 115, 41, 0.9)",
    80: "rgba(242, 115, 41, 0.8)",
    70: "rgba(242, 115, 41, 0.7)",
    60: "rgba(242, 115, 41, 0.6)",
    50: "rgba(242, 115, 41, 0.5)",
    40: "rgba(242, 115, 41, 0.4)",
    30: "rgba(242, 115, 41, 0.3)",
    20: "rgba(242, 115, 41, 0.2)",
    10: "rgba(242, 115, 41, 0.1)",
    5: "rgba(242, 115, 41, 0.05)",
  },
  db: {
    900: "#181B23",
    800: "#1F2029",
    700: "#353646",
    600: "#4B4D63",
    500: "#616480",
    400: "#797D9A",
    300: "#9699B0",
    200: "#B3B5C6",
    100: "#D1D2DC",
    50: "#EEEEF2",
  },
  gray: {
    900: "#171923",
    800: "#1A202C",
    700: "#2D3748",
    600: "#4A5568",
    500: "#718096",
    400: "#A0AEC0",
    300: "#CBD5E0",
    200: "#E2E8F0",
    100: "#EDF2F7",
    50: "#F7FAFC",
  },
  success: {
    dark: "#28a767",
    main: "#3cbe7d",
  },
  error: {
    main: "#F44336",
  },
  warning: {
    main: "#d9d10b",
  },
  info: {
    main: "#4559c9",
    light: "#899ed1",
  },
  background: {
    default: "#F7FAFC",
    paper: "#FFFFFF",
    darkPaper: "#c4c4c4",
    dark: "#1A202C",
    divider: "#e2e2e2",
    border: "#e2e2e2",
    lightGray: "#F3F5F8",
    box: "#F7FAFC",
    disabled: "#ddd",
  },
  text: {
    dark: "#000",
    main: "#2f3645",
    medium: "#424c61",
    light: "#808080",
    lightest: "#c4c4c4",
    label: "#808080",
  },
};

export default palette;
