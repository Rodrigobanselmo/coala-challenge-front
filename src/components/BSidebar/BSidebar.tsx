import Drawer from "@mui/material/Drawer";

import { sectionsDrawer } from "@/constants/drawer.constants";
import { SideBarNav } from "./SideBarNav";
import { StyledFlex } from "./styles";

export function BSidebar({
  isOpen,
  close,
  isTablet,
}: {
  isOpen: boolean;
  isTablet: boolean;
  close: () => void;
}): JSX.Element {
  return (
    <>
      {isTablet ? (
        <Drawer
          open={isOpen}
          onClose={close}
          ModalProps={{
            keepMounted: false,
          }}
        >
          <StyledFlex is_close={!isOpen ? 1 : 0} as="aside">
            <SideBarNav sections={sectionsDrawer} />
          </StyledFlex>
        </Drawer>
      ) : (
        <StyledFlex is_close={!isOpen ? 1 : 0} as="aside">
          <SideBarNav sections={sectionsDrawer} />
        </StyledFlex>
      )}
    </>
  );
}
