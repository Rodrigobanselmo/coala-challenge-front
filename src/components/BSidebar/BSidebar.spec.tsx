import React from "react";
import { render } from "@testing-library/react";
import { BSidebar } from "./BSidebar";
import { ThemeProvider } from "@emotion/react";
import theme from "@/configs/theme";
import { ThemeProvider as EmotionProvider } from "@emotion/react";

jest.mock("next/router", () => {
  let currentAsPath = "/";
  return {
    useRouter: () => ({
      asPath: currentAsPath,
    }),
  };
});

describe("BSidebar", () => {
  it("matches the snapshot with isOpen=true and isTablet=false", () => {
    const { container } = render(
      <EmotionProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <BSidebar isTablet={false} isOpen={true} close={() => {}} />
        </ThemeProvider>
      </EmotionProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot with isOpen=false and isTablet=true", () => {
    const { container } = render(
      <EmotionProvider theme={theme}>
        <ThemeProvider theme={theme}>
          <BSidebar isTablet={true} isOpen={false} close={() => {}} />
        </ThemeProvider>
      </EmotionProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
