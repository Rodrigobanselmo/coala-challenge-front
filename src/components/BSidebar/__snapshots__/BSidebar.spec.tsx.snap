// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BSidebar matches the snapshot with isOpen=false and isTablet=true 1`] = `<div />`;

exports[`BSidebar matches the snapshot with isOpen=true and isTablet=false 1`] = `
<div>
  <aside
    class="css-1o981aa"
    is_close="0"
  >
    <div
      class="MuiBox-root css-1daogsy"
    >
      <div
        class="MuiStack-root css-1qz3af0-MuiStack-root"
      >
        <div
          class="MuiStack-root css-961wsz-MuiStack-root"
        >
          <svg
            class="css-fw9wz5"
            height="1rem"
            viewBox="0 0 640 640"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <defs>
              <path
                d="m266.7 132.19 262.77 167.62L266.7 20.93v111.26Zm147.09 306.15 153.47-97.71-437-278.62v164.03c.14 4.8.83 9.51 2.07 14.14 1.25 4.63 3 9.05 5.29 13.27 2.27 4.22 5.01 8.12 8.2 11.71 3.19 3.57 6.75 6.73 10.69 9.49 6.82 4.34 40.94 26.08 102.37 65.19l154.91 98.5ZM168.78 398.4c-4.29 2.8-37.46 24.72-38.51 60.74v159.93L400.85 446.9 247.1 348.84c-41.77 26.43-67.88 42.95-78.32 49.56Z"
                id="a"
              />
            </defs>
            <use
              fill="#F27329"
              xlink:href="#a"
            />
            <use
              fill-opacity="0"
              stroke="#000"
              stroke-opacity="0"
              xlink:href="#a"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft MuiTypography-noWrap css-1l9enbr-MuiTypography-root"
          >
            Book
            <span
              class="MuiTypography-root MuiTypography-body1 css-kgpo3t-MuiTypography-root"
            >
              Swap
            </span>
          </p>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInfo MuiIconButton-sizeMedium css-43eudz-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kic248-MuiSvgIcon-root-MuiIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-19a83ym"
      >
        <div
          class="MuiStack-root css-1t035bm-MuiStack-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-30iklk"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-r459dm-MuiTypography-root"
              >
                Livros
              </p>
            </div>
            <div
              class="MuiStack-root css-1da3byd-MuiStack-root"
            >
              <div
                class="MuiBox-root css-9whsf3"
              >
                <a
                  href="/"
                >
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1k6yrop-MuiTypography-root-MuiLink-root"
                    is_active="1"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-qrqti-MuiSvgIcon-root-MuiIcon-root"
                      data-testid="AutoStoriesIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
                      />
                    </svg>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-8lqm6b-MuiTypography-root"
                      >
                        Livros para Troca
                      </p>
                    </div>
                  </a>
                </a>
              </div>
              <div
                class="MuiBox-root css-9whsf3"
              >
                <a
                  href="/seus-livros"
                >
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1c53box-MuiTypography-root-MuiLink-root"
                    is_active="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-qrqti-MuiSvgIcon-root-MuiIcon-root"
                      data-testid="LocalLibraryIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
                      />
                    </svg>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-8lqm6b-MuiTypography-root"
                      >
                        Seus Livros
                      </p>
                    </div>
                  </a>
                </a>
              </div>
              <div
                class="MuiBox-root css-9whsf3"
              >
                <a
                  href="/suas-trocas"
                >
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1c53box-MuiTypography-root-MuiLink-root"
                    is_active="0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-qrqti-MuiSvgIcon-root-MuiIcon-root"
                      data-testid="PublishedWithChangesIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m17.66 9.53-7.07 7.07-4.24-4.24 1.41-1.41 2.83 2.83 5.66-5.66 1.41 1.41zM4 12c0-2.33 1.02-4.42 2.62-5.88L9 8.5v-6H3l2.2 2.2C3.24 6.52 2 9.11 2 12c0 5.19 3.95 9.45 9 9.95v-2.02c-3.94-.49-7-3.86-7-7.93zm18 0c0-5.19-3.95-9.45-9-9.95v2.02c3.94.49 7 3.86 7 7.93 0 2.33-1.02 4.42-2.62 5.88L15 15.5v6h6l-2.2-2.2c1.96-1.82 3.2-4.41 3.2-7.3z"
                      />
                    </svg>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-8lqm6b-MuiTypography-root"
                      >
                        Suas Trocas
                      </p>
                    </div>
                  </a>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
